oligo_designer_toolsuite/oligo_specificity_filter/_filter_blastn.py

```diff
@@ -187,15 +198,171 @@ def _find_hits(
+        :type region_id: str
+        :return: Reference sequences
+        :rtype: list
+        """
+        # required_fields = [
+        #     "query",
+        #     "reference",
+        #     "alignment_length",
+        #     "query_start",
+        #     "query_end",
+        #     "query_length",
+        #     "query_sequence",
+        #     "reference_start",
+        #     "reference_end",
+        #     "reference_sequence",
+        #     "reference_strand"
+        # ]
+        # if not all(field in self.names_search_output for field in required_fields):
+        #     raise ValueError(
+        #         f"Some of the required fields {required_fields} are missing in the search results."
+        #     )
+        
+        # set the positions to a 0-based index
+        table_hits["query_start_corr"] = table_hits["query_start"] - 1  
+        table_hits["reference_start_corr"] = np.where(table_hits["reference_strand"] == "plus", table_hits["reference_start"] - 1, table_hits["reference_start"])  
+        table_hits["reference_end_corr"] = np.where(table_hits["reference_strand"] == "minus", table_hits["reference_end"] - 1, table_hits["reference_end"]) 
+
+        # Calculate adjusted "start" and "end" for BED format based on strand
+        table_hits["start"] = np.where(table_hits["reference_strand"] == "plus",
+                                        table_hits["reference_start_corr"] - table_hits["query_start_corr"],
+                                        table_hits["reference_end_corr"] - (table_hits["query_length"] - table_hits["query_end"]))
+        table_hits["end"] = np.where(table_hits["reference_strand"] == "plus",
+                                        table_hits["reference_end_corr"] + (table_hits["query_length"] - table_hits["query_end"]),
+                                        table_hits["reference_start_corr"] + table_hits["query_start_corr"])
+
+        # create bed file
+        bed = pd.DataFrame({  
+            "chr": table_hits["reference"],  
+            "start": table_hits["start"],  
+            "end": table_hits["end"],  
+            "name": table_hits["query"],  
+            "score": 0,  
+            "strand": table_hits["reference_strand"].map({"plus": "+", "minus": "-"})  
+        })
+        # adjust for possible overflows (e.g. new coordinates are not included in the gene boundaries)
+        # additionally we store how muchpadding we have to do to have two seqeunces of the same length
+        bed["overflow_start"] = bed["start"].apply(lambda x: -x if x < 0 else 0)
+        bed["start"] = bed["start"].apply(lambda x: x if x >= 0 else 0)
+        # what about having the leght of a region as a method of the reference database?
+        file_reference = reference_database.write_database_to_fasta(
+            filename="reference_db"
+        )
+        records = SeqIO.index(file_reference, "fasta")
+        bed["len_region"] = bed["chr"].apply(lambda x: len(records[x].seq))
```

Comment:
Might be faster to have something like:
`region_lengths = {region: len(record.seq) for region, record in records.items()}`
Then
`bed["len_region"] = bed["chr"].map(region_lengths)`